edk2toolext/rust_environment.py

```diff
@@ -0,0 +1,442 @@
+# @file rust_environment.py
+#
+# Helpers to check that Rust tools are present needed to compile Rust code
+# during firmare build.
+#
+# This functionality can be used to provide faster, direct feedback to a
+# developer about the changes they may need to make to successfully build Rust
+# code. Otherwise, the build will fail much later during firmware code
+# compilation when Rust tools are invoked with messages that are ambiguous or
+# difficult to find.
+#
+# Note:
+#   - Individual tools can be opted out by setting the environment variable
+#     `RUST_ENV_CHECK_TOOL_EXCLUSIONS` with a comma separated list of the tools
+#     to exclude. For example, "rustup, cargo tarpaulin" would not require that
+#     those tools be installed.
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+"""Rust Environment Helper Functions.
+
+Functions to help check that a build environment is ready to build Rust code.
```

Comment:
Everything in the `""" ... """` is what shows up in the documentation at the top of the page (in API Reference) if you think any of the additional information you currently have in the above `#` documentation would be useful to a normal user.

Reply:
Thanks, I forgot about that and inverted most of the content.